eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Snippets/ResourceMethodSnippets.cs

```diff
@@ -72,6 +72,39 @@ public static MethodBodyStatement CreateRequestContext(
             return Declare("context", typeof(RequestContext), New.Instance(typeof(RequestContext), requestContextParams), out contextVariable);
         }
 
+        public static MethodBodyStatement CreateUriFromMessage(VariableExpression messageVariable, out VariableExpression uriVariable)
+        {
+            // Uri uri = message.Request.Uri;
+            return Declare(
+                "uri",
+                typeof(RequestUriBuilder),
+                messageVariable.Property("Request").Property("Uri"),
+                out uriVariable);
+        }
+
+        public static MethodBodyStatement CreateRehydrationToken(
+            VariableExpression uriVariable,
```

Comment:
if possible, we could use `ScopedApi<Uri>` as the type here to make it strongly typed.